pkg/tablebackup/tablebackup.go

```diff
@@ -163,66 +163,66 @@ func (t *TableBackup) SaveRawMessage(msg []byte, lsn uint64) (uint64, error) {
 	return ln, nil
 }
 
-// the archiver goroutine is responsible for moving complete files to the final backup directory, as well as for closing
-// files in the temporary directory that didn't receive any writes for the specific amount of time (currently 3h)
-func (t *TableBackup) archiver() {
-	closeCall := time.NewTicker(archiverCloseNapTime)
-	q := queue.New(t.ctx)
-
-	// worker routine to do the heavy-lifting
-	go func() {
-		for {
-			select {
-			case <-time.After(archiverWorkerNapTime):
-				for {
-					obj, err := q.Get()
-					if err != nil {
-						if err == context.Canceled {
-							return
-						}
-						// the only other error we expect is that the queue is empty, in which case we go to sleep
-						break
+// the archiveCloseFiles goroutine is responsible for closing files in the temporary directory
+// that didn't receive any writes for the specific amount of time (currently 3h)
+func (t *TableBackup) archiverCloseFiles() {
```

Comment:
I think you've got the names mixed: the routine that you've named archiverCloseFiles is actually the one that does the archiving itself. 

I think your changes are good: the archiver worker code should be moved from the anonymous nested function into a separate, named one; both the routine now listen to the context done, as they should. However, I think we should name the routine that closes files somehow differently, without the word archiver in it, i.e. janitor :-) to avoid naming confusion (or make the archiverCloseFiles run inside the main archiver goroutine).